### DISCUSSION TITLE
CORS Error on Player Only using Public src URL

### DISCUSSION THREAD

**Discussion by Seven11HY** [muxinc/next-video#322] [Q&A] [answered]
When I use the `Player from 'next-video/player'` and insert the `src` to my video.mp4 hosted in my S3 Bucket and delivered via Cloudfront CDN, I always get a CORS error. 

<img width="548" alt="Screenshot 2024-08-28 at 12 49 31" src="https://github.com/user-attachments/assets/e671a407-eff3-4143-8bf6-fab207a6a273">

And the Player throughs the Error:
`[mux-player 2.9.0] MediaError: MEDIA_ELEMENT_ERROR: Format error`

If I use the same URL in the normal HTML5 video tag, it works without a problem.

**Comment by mmcc** [comment]
Hm that's odd... I'll dig in more, but we've been talking about swapping out Mux Player for something that's a better fit for this use case given that we need to support more than Mux assets here. This prompted me to go ahead and [open an issue for tracking that work](https://github.com/muxinc/next-video/issues/292).

**Comment by luwes** [comment, marked as answer]
could you try adding the `crossOrigin` prop to the Player component?

**Comment by howells** [comment]
FWIW, I get the CORS issue using the Player, but not the BackgroundVideo
